src/ui/public/visualize/visualization.js

```diff
@@ -114,16 +121,25 @@ uiModules
           renderSubscription.unsubscribe();
         });
 
-        if (!$scope.vis.visualizeScope) {
-          $scope.$watchGroup(['visData', 'vis.params'], () => {
+        // Listen on changes to trigger a render if listenOnChange is not false
+        // i.e. true or has not been used (undefined in that case)
+        if ($scope.listenOnChange !== false) {
+          const onChangeListener = () => {
             $scope.$emit('render');
-          });
+          };
 
-          resizeChecker.on('resize',  () => {
-            $scope.$emit('render');
+          $scope.$watchGroup(['visData', 'vis.params'], onChangeListener);
+
+          $scope.uiState.on('change', onChangeListener);
```

Comment:
Listen to `uiState` changes in `visualization`. Earlier we just listened to it in `visualize` and the `vislib` visualizations. This was actually a bug, which would have caused charts not to update, if you use `visualization` directly (withou `visualize` around it) for a non vislib chart and change the `uiState`. This will now properly trigger a render of the visualization.